tests/unit/yieldables-test.js

```diff
@@ -111,3 +111,114 @@ test("all cancels all joined tasks if parent task is canceled", function(assert)
   assert.equal(childTask.get('concurrency'), 0);
 });
 
+test("allSettled behaves like Promise.allSettled", function(assert) {
+  assert.expect(6);
+
+  let defers = [];
+  let Obj = Ember.Object.extend({
+    parent: task(function * () {
+      let task = this.get('child');
+      let allPromise = allSettled([
+        task.perform(),
+        task.perform(),
+        task.perform(),
+      ]);
+      assert.equal(typeof allPromise.then, 'function');
+      let values = yield allPromise;
+      assert.deepEqual(values, [{ state: 'fulfilled', value: 'a' }, { state: 'fulfilled', value: 'b' }, { state: 'fulfilled', value: 'c' }]);
```

Comment:
Here is the full log of the values, the stack and line are under the reason, thats why I built out the other solution to re-construct reason object. https://cl.ly/1c1E2u0w1H3O